relay/relay.py

```diff
@@ -0,0 +1,364 @@
+    """Function decorator for REST API methods which validates the relay secret
+
+    This decorator wraps the supplied API method callback with a function which
+    checks that the secret is valid before calling the method.
+
+    Args:
+        func: the API method route callback function
+
+    Returns:
+        A function which validates the secret, calls the provided method
+        callback function, and returns the value that it returns.
+    """
+
+    @functools.wraps(func)
+    def do_validation(secret: str, *args, **kwargs) -> HTTPResponse:
+        """Wrapper function which validates the relay secret
+
+        If the "secret" (i.e., the first URI path parameter, which is the first
+        component in the API route) matches the deployment configuration, then
+        invoke the API method function and return its result; otherwise
+        return a FORBIDDEN response instead of calling the API method.
+
+        The secret argument is omitted from the call to the method, since it is
+        only used for validation.
+
+        Args:
+            secret:  the contents of the first URI path parameter
+            All other arguments are passed through to the API method callback.
+
+        Returns:
+            If the secret is valid, returns the response returned by the wrapped
+            method function; otherwise, returns a FORBIDDEN response (or, as a
+            special case, returns NOT_FOUND if the request is for the favicon).
+        """
+        if secret == click.get_current_context().meta[CTX_SECRET]:
+            return func(*args, **kwargs)
+
+        # Special case this request which seems to come from testing with the
+        # browser, just to quiet the noise.
+        if secret == "favicon.ico":
+            return HTTPResponse(status=HTTPStatus.NOT_FOUND)
+
+        logging.warning(
+            'Secret validation failed:  expected "%s", got "%s"',
+            click.get_current_context().meta[CTX_SECRET],
+            secret,
+        )
+        return HTTPResponse(status=HTTPStatus.FORBIDDEN)
+
+    return do_validation
+
+
+@app.get("/<secret>")
+@validate_secret
+@click.pass_context
+def relay_status(context: click.Context) -> HTTPResponse:
+    """Relay server status API
+
+    Args:
+        context:  the Click context object
+                    - for access to the local files directory path
+
+    Returns:
+        An HTTP response with a status of OK and a JSON payload containing
+        status information (currently, the output from `ls` listing the files
+        in the upload directory).
+    """
+    logging.info("request to report status")
+    body = {"disk utilization": get_disk_utilization_str(context.meta[CTX_DIRECTORY])}
+
+    cp = subprocess.run(
+        ["ls", "-l"], cwd=context.meta[CTX_DIRECTORY], capture_output=True, text=True
+    )
+    if cp.returncode:
+        body["error"] = cp.stderr.strip()
+    else:
+        body["files"] = cp.stdout.strip().split("\n")
```

Comment:
Yeah...this function started out with a static string saying the equivalent of "Yeah, I'm here!".  Then I decided (at least for testing) that it would be handy to have the file list.  Then I noted that the Pbench Server upload reports the disk statistics, and I thought that that was cool (and simple).

But, reporting out the list of files is a small violation of the unguessibility principle, but only among people who have the server ID...so I'm ambivalent about retaining this feature.

But, yes, if we wanted to keep this method returning a list of the available files, then it would presumably be better to do more than just to return the `ls` output...but I figured that this was something that we could consider and iterate on.

Reply:
Hopefully the outcome of Discussion #3414 will resolve this.